spec/lib/websocket_proxy_spec.rb

```diff
@@ -8,9 +36,20 @@
   subject { described_class.new(env, console, logger) }
 
   describe '#initialize' do
+    before do
+      allow(TCPSocket).to receive(:open) # prevent real sockets from opening
+    end
+
     it 'sets the URL' do
       expect(subject.url).to eq("ws://#{host}#{uri}")
     end
+
+    describe 'picks expected protocol and driver' do
+      it_behaves_like('picking protocol', 'vnc', WebsocketSocket, ['binary'], true)
+      it_behaves_like('picking protocol', 'spice', WebsocketSocket, ['binary'], true)
+      it_behaves_like('picking protocol', 'webmks', WebsocketWebmks, ['binary'], true)
+      it_behaves_like('picking protocol', 'webmks-uint8utf8', WebsocketWebmksUint8utf8, ['uint8utf8'], false)
+    end
```

Comment:
I'd rather prefer `include_examples` over `it_behaves_like`.

Reply:
Also having a hash as arguments can be much more descriptive here...

Reply:
Agreed. Done.